dashboard/src/components/Catalog/Catalog.tsx

```diff
@@ -234,123 +311,145 @@ function Catalog(props: ICatalogProps) {
+              )}
+              {allRepos.length > 0 && (
+                <div className="filter-section">
+                  <label>Application Repository</label>
+                  <FilterGroup
+                    name={filterNames.REPO}
+                    options={allRepos}
+                    currentFilters={filters[filterNames.REPO]}
+                    onAddFilter={addFilter}
+                    onRemoveFilter={removeFilter}
+                  />
+                </div>
+              )}
+              {allProviders.length > 0 && (
+                <div className="filter-section">
+                  <label className="filter-label">Operator Provider</label>
+                  <FilterGroup
+                    name={filterNames.OPERATOR_PROVIDER}
+                    options={allProviders}
+                    currentFilters={filters[filterNames.OPERATOR_PROVIDER]}
+                    onAddFilter={addFilter}
+                    onRemoveFilter={removeFilter}
                   />
+                </div>
+              )}
+            </div>
+          </Column>
+          <Column span={10}>
+            <>
+              <div className="filter-summary">
+                {Object.keys(filters).map(filterName => {
+                  if (filters[filterName].length) {
+                    return filters[filterName].map((filterValue: string) =>
+                      filterValue.length ? (
+                        <span key={`${filterName}-${filterValue}`} className="label label-info">
+                          {filterName}: {filterValue}{" "}
+                          <CdsIcon
+                            shape="times"
+                            onClick={removeFilterFunc(filterName, filterValue)}
+                          />
+                        </span>
+                      ) : (
+                        ""
+                      ),
+                    );
+                  }
+                  return null;
+                })}
+              </div>
+              <div className="catalogContainer">
+                <Row>
+                  <>
+                    <CatalogItems
+                      charts={filteredCharts}
+                      csvs={filteredCSVs}
+                      cluster={cluster}
+                      namespace={namespace}
+                    />
+                    {!hasFinishedFetching &&
+                      (!filters[filterNames.TYPE].length ||
+                        filters[filterNames.TYPE].find((type: string) => type === "Charts")) && (
+                        <div className="endPageMessage">
+                          <LoadingWrapper loaded={false} />
+                          <span>
+                            {!charts.length && !csvs.length
+                              ? "Loading catalog..."
+                              : "Scroll down to discover more applications"}
+                          </span>
+                        </div>
+                      )}
```

Comment:
two minor things here:
 - I would use the small loading spinner, I think there is a property for that
 - Would remove the messages (including "no remaining applications"), it's clear that it's loading when the loading wrapper is shown and it's weird when using filters. For example if I search "wordpress" I see the message "scroll down..." but there is no need to scroll down, is just a matter of letting it load.

Reply:
- I'll test the small spinner out, I remembered to have seen this prop and will add some screenshots for us to compare.
- I'm not sure about that, searching for WordPress perhaps results in a few charts (and "no remaining applications" won't be displayed if searching). However, search results can possibly be paginated, so these messages do make sense for me. Again, I will change it and will post here some gifs for us to decide.